ci/matrix.yaml

```diff
@@ -1,384 +1,284 @@
-  'test_gpu'   : { 'prefix': 'test', 'args': '-gpu-only' }
-  'test_nolid' : { 'prefix': 'test', 'args': '-no-lid' }
-  'test_lid0'  : { 'prefix': 'test', 'args': '-lid0' }
-  'test_lid1'  : { 'prefix': 'test', 'args': '-lid1' }
-  'test_lid2'  : { 'prefix': 'test', 'args': '-lid2' }
-
-# When a listed project has a `test` job, it will be replaced with the specified list of finer-grain jobs.
-project_expanded_tests:
-  'thrust' : ['test_gpu', 'test_cpu']
-  'cub'    : ['test_nolid', 'test_lid0', 'test_lid1', 'test_lid2']
-
-# Human readable name for jobs. Default behavior is to capitalize the first letter.
-formatted_jobs:
-  'nvrtc':          'NVRTC'
-  'verify_codegen': 'VerifyCodegen'
-  'test_cpu':       'TestCPU'
-  'test_gpu':       'TestGPU'
-  'test_nolid':     'TestGPU'
-  'test_lid0':      'HostLaunch'
-  'test_lid1':      'DeviceLaunch'
-  'test_lid2':      'GraphCapture'
-
-# Human readable name for projects. Default behavior uses the project name as-is.
-formatted_project_names:
-  'libcudacxx': 'libcu++'
-  'cub':        'CUB'
-  'thrust':     'Thrust'
-  'cccl':       'CCCL'
-
-# Human readable name for compilers. Default behavior uses the "compiler.name" tag as-is.
-formatted_cxx_names:
-  'llvm':   'clang'
-  'oneapi': 'Intel'
-  'cl':     'MSVC'
-
-# All known GPUs
+# The version of the devcontainer images to use from https://hub.docker.com/r/rapidsai/devcontainers
+devcontainer_version: '24.08'
+
+# All supported C++ standards:
+all_stds: [11, 14, 17, 20]
+
+ctk_versions:
+  11.1: { stds: [11, 14, 17,   ] }
+  11.8: { stds: [11, 14, 17,   ] }
+  12.0: { stds: [11, 14, 17, 20] }
+  12.4: { stds: [11, 14, 17, 20], aka: 'curr' }
+
+device_compilers:
+  nvcc: # Version / stds are taken from CTK
+    name: 'nvcc'
+    exe: 'nvcc'
+  clang: # Requires cxx=clang. Version / stds are taken from cxx compiler.
+    name: "ClangCUDA"
+    exe: 'clang++'
+
+host_compilers:
+  gcc:
+    name: 'GCC'
+    container_tag: 'gcc'
+    exe: 'g++'
+    versions:
+      6:  { stds: [11, 14,       ] }
+      7:  { stds: [11, 14, 17,   ] }
+      8:  { stds: [11, 14, 17,   ] }
+      9:  { stds: [11, 14, 17,   ] }
+      10: { stds: [11, 14, 17, 20] }
+      11: { stds: [11, 14, 17, 20] }
+      12: { stds: [11, 14, 17, 20] }
+      13: { stds: [11, 14, 17, 20] }
+  clang:
+    name: 'Clang'
+    container_tag: 'llvm'
+    exe: 'clang++'
+    versions:
+      9:  { stds: [11, 14, 17,   ] }
+      10: { stds: [11, 14, 17,   ] }
+      11: { stds: [11, 14, 17, 20] }
+      12: { stds: [11, 14, 17, 20] }
+      13: { stds: [11, 14, 17, 20] }
+      14: { stds: [11, 14, 17, 20] }
+      15: { stds: [11, 14, 17, 20] }
+      16: { stds: [11, 14, 17, 20] }
+      17: { stds: [11, 14, 17, 20] }
+  msvc:
+    name: 'MSVC'
+    container_tag: 'cl'
+    exe: cl
+    versions:
+      14.16: { stds: [    14,       ], aka: '2017' }
+      14.29: { stds: [    14, 17,   ], aka: '2019' }
+      14.36: { stds: [    14, 17, 20]              }
```

Comment:
I'm not sure 14.36 exists in the devcontainers. I'll go poke around. We'll need to update to 14.40 soon.

Reply:
https://github.com/rapidsai/devcontainers/blob/78b51603afa554f5a3f80628094e592acb1b659f/matrix.yml#L129C106-L130C106

Ahhh

Reply:
It's only used for the CTK12.0 images that cudax will need once we enable those builds. Not sure if it was intentionally used there or if it just missed an upgrade.